### PR TITLE
Add an option for building SDL3 from source without X11

### PR DESCRIPTION
**Pull request by hansl**
My project uses SDL3 on an embedded platform without X11, so I came into this error. This fix seems to work locally.

### CODE REVIEW

**Comment by maia-s** [comment]
Thank you! I'd like to name the feature ~~`unix-console-build`~~ instead, to match the SDL variable. If you'd like you can update the PR, or I can make the change.
edit: `sdl-unix-console-build`

**Comment by maia-s** [comment]
Actually I was considering using a `build-` prefix for build related features for namespacing, but `build-unix-console-build` is kind of a mouthful. What do you think?

**Comment by Lokathor** [comment]
if it can match exactly with a real sdl3 feature then I'd use the matching name

**Comment by maia-s** [comment]
The full feature name from SDL's cmake script is `SDL_UNIX_CONSOLE_BUILD`. The SDL prefix is kind of implied, but it *would* serve nicely as a namespace (better than `build`, now that I think about it): `sdl-unix-console-build`. I'm not sure if that's what you meant tho

The namespacing would be more useful for some other features, like `SDL_LIBC` which I'd like to add support for (disabling it in that case). Naming that `libc` would be confusing. If some features need it,    namespacing all the build features the same makes it consistent

**Comment by Lokathor** [comment]
`sdl-unix-console-build` would be a perfectly fine name

**Comment by maia-s** [comment]
Great, let's go with that then 😊

File: sdl3-sys/Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,10 @@ link-static = []
 # Link SDL as a mac framework. The link-static feature has no effect if this is enabled.
 link-framework = []
 
+# When building from source, use the `SDL_UNIX_CONSOLE_BUILD` flag to disable the X11 dependency
+# and build a console-only version of SDL. This has no effect when not building from source.
+console-build = []
```
Comment (maia-s) [inline sdl3-sys/Cargo.toml]: Can you please rename this feature to `sdl-unix-console-build`? Thanks!

**Review by maia-s — CHANGES_REQUESTED** [review]

**Comment by hansl** [comment]
@maia-s done. PTAL.

File: sdl3-sys/Cargo.toml (hunk 1 above)
Comment (hansl) [reply sdl3-sys/Cargo.toml]: Seems like I did the change while you were doing the review. Cheers!

**Comment by maia-s** [comment]
Thank you!

**Comment by maia-s** [comment]
I've published sdl3-sys 0.1.3 which includes this.

**Pull request merged**